packages/x-tree-view/src/internals/useTreeView/useTreeViewModels.ts

```diff
@@ -48,9 +48,7 @@ export const useTreeViewModels = <
 
   const models = Object.fromEntries(
     Object.entries(modelsRef.current).map(([modelName, model]) => {
-      const value = model.isControlled
-        ? props[modelName as keyof DefaultizedParams]
-        : modelsState[modelName];
+      const value = props[modelName as keyof DefaultizedParams] ?? modelsState[modelName];
```

Comment:
This change makes sure that if we switch a model from controlled to uncontrolled, we will still have a valid value on the model.
This is a bad pattern and a warning will be shown (defined below in this file), but at least it does not crash with in some random place of the code and people can understand what is the problem.